pylds/base_discrete.py

```diff
@@ -26,6 +27,109 @@ def check_if_points_escape_box(u, box_boundaries):
     u_indices = (x >= box_x_min) & (x <= box_x_max) & (y >= box_y_min) & (y <= box_y_max)
     return u_indices
 
+def pbc_correction_coords_single_axis(x, box_origin, box_length):
+    """
+    Correct coordinate in a single axis by Periodic Boundary Conditions (PBCs).
+
+    Parameters
+    ----------
+    x : array_like, shape(n, )
+        points in axis to correct by PBCs
+    
+    origin : array
+        
+    box_length : array
+        
+    Returns
+    -------
+    x_pbc : array_like, shape(n, )
+        array of corrected points for periodic box
+    """
+    x_pbc = x
+    x0 = box_origin
+    L = box_length
+    if not isinstance(x0, bool) or not isinstance(L, bool):
+        #apply PBC correction
+        x = x + L/2 - x0
+        x = np.mod(x + 2*L, L) 
+        x_pbc = x - L/2 + x0
+        
+    return x_pbc
+
+def pbc_correction_coords(u, periodic_boundaries):
+    """
+    Correct coordinates in 2D plane by Periodic Boundary Conditions (PBCs).
+
+    Parameters
+    ----------
+    u : array_like, shape(n, )
+        points in plane to correct by PBCs
+    
+    periodic_boundaries : list of 2-tuples of floats
+        periodic box lower and upper limits along X and Y axes
+        
+    Returns
+    -------
+    u_pbc : array_like, shape(n, )
+        array of corrected points for periodic box
+    """
+    N_dims = len(periodic_boundaries)
+    pbc = periodic_boundaries
+    f = pbc_correction_coords_single_axis
+    u_pbc = [f(u.T[i], *pbc[i]) for i in range(N_dims)]
+    
+    return np.column_stack(u_pbc)
+
+def pbc_correction_distances_single_axis(dx, box_length):
+    """
+    Correct distances in a single axis by Periodic Boundary Conditions (PBCs).
+
+    Parameters
+    ----------
+    dx : array_like, shape(n, )
+        distances in axis to correct by PBCs
+        
+    box_length : float or bool
+        if no PBCs, this must be False (bool)
```

Comment:
If no periodic boundary conditions apply, will this function be called?

Reply:
Yes, it will return `x_pbc = x` (identity function). An alternative would be adding an `if`-`else` statement in `pbc_correction_coords` where looping is done

```
f = pbc_correction_coords_single_axis    
u_pbc = [f(u.T[i], *pbc[i]) for i in range(N_dims)]
```
But this seems a bit cumbersome, thought better if `pbc_correction_coords_single_axis` acting as an identity function if pbc `False` for one axis.

Reply:
ok